### PR TITLE
Extend CInt(Fix(x)) optimization to other whole-number-returning functions.

### LINKED ISSUE #26907: Simplify code for CInt(Math.Ceiling(x))
As a follow-on to the work for https://github.com/dotnet/roslyn/issues/25692, the following additional opportunities for code improvement exist:
1. `CInt(Math.Truncate(x))` (drop call to `Truncate` and to `Round`)
2. `CInt(Math.Ceiling(x))` (drop the call to `Round`)
3. `CInt(Math.Floor(x))` (drop the call to `Round`)
4. `CInt(Math.Round(x))` (drop a duplicate call to `Round`)
5. `CInt(Int(x))` (drop the call to `Round`)

Please modify the VB compiler to produce improved code for these scenarios.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #26907

@cston @jcouv @jaredpar This is a follow-up to code you reviewed yesterday.

### CODE REVIEW

**Comment by gafter** [comment]
@cston @jcouv @jaredpar Could you please have a look at this follow-up to last week's VB optimizations?

File: src/Compilers/Core/Portable/WellKnownMembers.cs [hunk 1]
```diff
@@ -2980,16 +2980,57 @@ static WellKnownMembers()
...
                     1,                                                                                                                              // Method Signature
-                    (byte)SignatureTypeCode.TypeHandle, (byte)SpecialType.System_Double,                                    // Return type
-                    (byte)SignatureTypeCode.TypeHandle, (byte)SpecialType.System_Double,                                    // Number As System.Double
+                    (byte)SignatureTypeCode.TypeHandle, (byte)SpecialType.System_Double, // Return type
+                    (byte)SignatureTypeCode.TypeHandle, (byte)SpecialType.System_Double, // Number As System.Double
+
+                 // Microsoft.VisualBasic.Conversion.Int(Number As System.Single)
+                (byte)(MemberFlags.Method | MemberFlags.Static),                                                                                    // Flags
+                 (byte)WellKnownType.ExtSentinel, (byte)(WellKnownType.Microsoft_VisualBasic_Conversion - WellKnownType.ExtSentinel),               // DeclaringTypeId
```
Comment (jcouv) [inline src/Compilers/Core/Portable/WellKnownMembers.cs]: Nit: the indentation on the "DeclaringTypeId" line is off by one (including the two from the original PR).

File: src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb [hunk 2]
```diff
@@ -1339,20 +1343,50 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
+            If "Fix".Equals(methodName) Then
+                Select Case node.Type.SpecialType
+                    Case SpecialType.System_Single
+                        Return node.Method = Me.Compilation.GetWellKnownTypeMember(WellKnownMember.Microsoft_VisualBasic_Conversion__FixSingle)
+                    Case SpecialType.System_Double
+                        Return node.Method = Me.Compilation.GetWellKnownTypeMember(WellKnownMember.Microsoft_VisualBasic_Conversion__FixDouble)
+                    Case Else
+                        Return False
+                End Select
+            ElseIf "Truncate".Equals(methodName) Then
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb]: Would it be worth it to also add a check on special type (before checking the method), as a shortcut?

Comment (gafter) [reply src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.vb]: There is no way to go from a type to its special type. There is only such a thing for well-known types. The best we could do is look up the special type and see if the type is the same. That is as expensive as the test we already do (once the cache is filled).

---
In reply to: [190118936](https://github.com/dotnet/roslyn/pull/26948#discussion_r190118936) [](ancestors = 190118936)

File: src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/WellKnownTypeValidationTests.vb [hunk 3]
```diff
@@ -677,7 +677,9 @@ End Namespace
                          WellKnownMember.Microsoft_VisualBasic_Information__VbTypeName,
                          WellKnownMember.Microsoft_VisualBasic_Interaction__CallByName,
                          WellKnownMember.Microsoft_VisualBasic_Conversion__FixSingle,
-                         WellKnownMember.Microsoft_VisualBasic_Conversion__FixDouble
+                         WellKnownMember.Microsoft_VisualBasic_Conversion__FixDouble,
+                         WellKnownMember.Microsoft_VisualBasic_Conversion__IntSingle,
+                         WellKnownMember.Microsoft_VisualBasic_Conversion__IntDouble
```
[the remark this answers is not in the document]
Comment (gafter) [reply src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/WellKnownTypeValidationTests.vb]: I'm OK with doing that, but not as a part of this PR.

---
In reply to: [190119270](https://github.com/dotnet/roslyn/pull/26948#discussion_r190119270) [](ancestors = 190119270)

**Comment by AnthonyDGreen** [comment]
@gafter, you beautiful man!

**Pull request merged**